pandas/io/pytables.py

```diff
@@ -3247,7 +3247,7 @@ def validate_data_columns(self, data_columns, min_itemsize):
         # evaluate the passed data_columns, True == use all columns
         # take only valide axis labels
         if data_columns is True:
-            data_columns = axis_labels
+            data_columns = list(axis_labels)
```

Comment:
in fact this should be not done here at all. The ``data_columns`` should be cleaned in a central location (IIRC they already are and just needs an adjustment)